packages/astro/src/core/create-vite.ts

```diff
@@ -180,32 +179,28 @@ export async function createVite(
 	// We also need to filter out the plugins that are not meant to be applied to the current command:
 	// - If the command is `build`, we filter out the plugins that are meant to be applied for `serve`.
 	// - If the command is `dev`, we filter out the plugins that are meant to be applied for `build`.
-	if (command) {
-		let plugins = settings.config.vite?.plugins;
-		if (plugins) {
-			const { plugins: _, ...rest } = settings.config.vite;
-			const applyToFilter = command === 'build' ? 'serve' : 'build';
-			const applyArgs = [
-				{ ...settings.config.vite, mode },
-				{ command, mode },
-			];
-			// @ts-expect-error ignore TS2589: Type instantiation is excessively deep and possibly infinite.
-			plugins = plugins.flat(Infinity).filter((p) => {
-				if (!p || p?.apply === applyToFilter) {
-					return false;
-				}
-
-				if (typeof p.apply === 'function') {
-					return p.apply(applyArgs[0], applyArgs[1]);
-				}
+	if (settings.config.vite?.plugins) {
+		let { plugins, ...rest } = settings.config.vite;
+		const applyToFilter = command === 'build' ? 'serve' : 'build';
+		const applyArgs = [
+			{ ...settings.config.vite, mode },
+			{ command, mode },
+		];
+		// @ts-expect-error ignore TS2589: Type instantiation is excessively deep and possibly infinite.
+		plugins = plugins.flat(Infinity).filter((p) => {
+			if (!p || p?.apply === applyToFilter) {
+				return false;
+			}
 
-				return true;
-			});
+			if (typeof p.apply === 'function') {
+				return p.apply(applyArgs[0], applyArgs[1]);
+			}
 
-			result = vite.mergeConfig(result, { ...rest, plugins });
-		} else {
-			result = vite.mergeConfig(result, settings.config.vite || {});
-		}
+			return true;
+		});
+		result = vite.mergeConfig(result, { ...rest, plugins });
+	} else {
+		result = vite.mergeConfig(result, settings.config.vite || {});
```

Comment:
The main change in this diff is that previously when `command` is not set, it means it's in `astro sync`. However, it did not run `result = vite.mergeConfig(result, settings.config.vite || {});`, causing the issue.

We can't run that `mergeConfig` if `command` is undefined because that calls Vite plugins with `apply: 'serve'` in builds. So I made `astro sync` run in `build` mode only to fix this.